src/Servers/Kestrel/Core/src/Internal/Http2/Http2FrameWriter.cs

```diff
@@ -128,10 +128,10 @@ private async Task WriteToOutputPipe()
                     var buffer = readResult.Buffer;
 
                     // Check the stream window
-                    var (actual, remainingStream) = producer.ConsumeStreamWindow(buffer.Length);
+                    var actual = producer.ConsumeStreamWindow(buffer.Length);
 
                     // Now check the connection window
-                    (actual, var remainingConnection) = ConsumeConnectionWindow(actual);
+                    actual = ConsumeConnectionWindow(actual);
```

Comment:
Q: Are we leaking stream window bytes?
Setup: The stream has an available window of 50. The connection has an available window of 40. We try to write 50+ bytes. The steam window is consumed. The connection window is consumed, but then the buffer is sliced to 40 bytes. Are those 10 extra bytes credited back to the stream window for future writes?

Reply:
Good catch!

Reply:
Fixed. Please review.